src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -74,7 +74,8 @@ Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
         const SchedulingKey scheduling_key(
             task_spec.GetSchedulingClass(), task_spec.GetDependencyIds(),
             task_spec.IsActorCreationTask() ? task_spec.ActorCreationId()
-                                            : ActorID::Nil());
+                                            : ActorID::Nil(),
+            task_spec.GetRuntimeEnvHash(), task_spec.JobId());
```

Comment:
Can we confirm that it makes sense to add JobID here?

Reply:
Hmmm, probably best not to have job id since it doesn't really do anything. Workers belong to jobs, so job id will always be the same.